tests/uitools/qml_quick/functional_tests/FloorFilterControllerFuncTest/tst_floorfiltercontroller.qml

```diff
@@ -0,0 +1,189 @@
+                }
+            }
+        }
+    }
+
+    MapView {
+        id: viewSingle
+        Map {
+            id: mapSingle
+            item: PortalItem {
+                itemId: "f133a698536f44c8884ad81f80b6cfc7"
+                portal: Portal {
+                    Component.onCompleted: load()
+                    onErrorChanged: {
+                        if (loadStatus === Enums.LoadStatusFailedToLoad) {
+                            console.log(error.message);
+                        }
+                    }
+                    url: "https://www.arcgis.com"
+                }
+            }
+        }
+    }
+
+    // 1 site, 1 facility, 3 levels
+    Component {
+        id: controlSingleComponent
+        FloorFilterController {
+            geoView: viewSingle
+        }
+    }
+
+    // more sites, more facilities
+    Component {
+        id: controlMultiComponent
+        FloorFilterController {
+            geoView: viewMulti
+        }
+    }
+
+    SignalSpy {
+        id: ffLoadedSpy
+        signalName: "onLoaded" // signal outputted from FloorFilter once the floorManager has been loaded
+    }
+
+    function loadFloorManagerSingle() {
+        ffLoadedSpy.clear();
+        var control = createTemporaryObject(controlSingleComponent, floorFilterControllerFuncTest);
+
+        if(!control.floorManager || control.floorManager.loadStatus !== Enums.LoadStatusLoaded) {
+            ffLoadedSpy.target = control;
+            ffLoadedSpy.wait(10000);
+        }
+        compare(control.floorManager.loadStatus, Enums.LoadStatusLoaded);
+        return control;
+    }
+
+    function loadFloorManagerMultiple() {
+        ffLoadedSpy.clear();
+        var control = createTemporaryObject(controlMultiComponent, floorFilterControllerFuncTest);
+
+        if(!control.floorManager || control.floorManager.loadStatus !== Enums.LoadStatusLoaded) {
+            ffLoadedSpy.target = control;
+            ffLoadedSpy.wait(10000);
+        }
+        compare(control.floorManager.loadStatus, Enums.LoadStatusLoaded);
+        return control;
+    }
+
+    function test_allSitesFacilitySelection() {
+        var control = loadFloorManagerMultiple();
+        // autofires populateAllFacilities()
+        control.selectedSiteRespected = false;
+        var facility = control.floorManager.facilities[0];
+        verify(facility.facilityId);
+        control.setSelectedFacilityId(facility.facilityId);
+        compare(control.selectedSiteId, facility.site.siteId);
+        compare(control.selectedSite.siteId, facility.site.siteId);
+    }
+
+    // singlesite mode on. only 1 site and 1 facility, automatically select them on load.
+    function test_singleSiteAutoSelect() {
+        var control = loadFloorManagerSingle();
+        verify(1, control.floorManager.sites.count);
+        compare(1, control.sites.count);
+        compare(1, control.facilities.count);
+        verify(control.selectedSiteId) // not empty string
```

Comment:
I'd be more comfortable comparing to an empty string than relying on implicit conversion here. `false == "0"` in javascript, which is the sort of thing which makes me nervous.